weather_dl/download_pipeline/pipeline.py

```diff
@@ -133,9 +133,20 @@ def _create_partition_config(option: t.Tuple, config: t.Dict) -> t.Dict:
         copy[key] = [option[idx]]
 
     out['selection'] = copy
+    if index != -1:
+        out['parameters']['__index__'] = index
+
     return out
 
 
+def partition_by_index(config: t.Dict, num_partitions: int) -> int:
+    """Split config partitions into groups based on their `__index__` value."""
+    index = config.get('parameters', {}).get('__index__', -1)
+    if index == -1:
+        return index
+    return index % num_partitions
```

Comment:
Should there be a check that index is >0? Currently it's created using enumerate, so it has to be, but just in case.

Reply:
So, as long as `num_partitions` is positive, the resulting mod operation will also be positive: 
> The modulo operator always yields a result with the same sign as its second operand (or zero)
https://docs.python.org/3/reference/expressions.html#binary-arithmetic-operations

However, it's possible that `num_partitions` receives a negative number. How about I update the documentation of the method to indicate that the number of partitions should be at least 1?